src/studio/src/designer/backend/Services/Implementation/RepositorySI.cs

```diff
@@ -1050,9 +1050,22 @@ private bool UpdateApplicationWithAppLogicModel(string org, string app, string d
                 application.DataTypes = new List<PlatformStorageModels.DataType>();
             }
 
-            PlatformStorageModels.DataType logicElement = application.DataTypes.Single(d => d.AppLogic != null);
+            PlatformStorageModels.DataType logicElement = application.DataTypes.SingleOrDefault(d => d.Id == dataTypeId);
+            if (logicElement == null)
+            {
+                logicElement = new PlatformStorageModels.DataType
+                {
+                    Id = dataTypeId,
+
+                    // TODO: these properties
+                    // TaskId = "Task_1",
+                    AllowedContentTypes = new List<string>() { "application/xml" },
+                    MaxCount = 1,
+                    MinCount = 1,
+                };
+                application.DataTypes.Add(logicElement);
```

Comment:
How does this affect the "default" element that is part of the app template? It would be unfortunate if that element is left hanging and unused.
Should we remove that from the template `applicationMetadata.json` file entirely? @TheTechArch what do you think?

Reply:
@everystone I suggest you remove the default dataType element from the template. `src/Altinn.Apps/AppTemplates/AspNet/App/config/applicationMetadata` and `src/studio/AppTemplates/AspNet/App/config/applicationMetadata`. Since your code is adding the element if it doesn't exist anyways.

Reply:
done